utils/appliance/db.py

```diff
@@ -0,0 +1,385 @@
+import attr
+from cached_property import cached_property
+import fauxfactory
+from textwrap import dedent
+
+from utils import db, conf, clear_property_cache, datafile
+from utils.log import logger_wrap
+from utils.path import scripts_path
+from utils.version import LATEST
+from utils.wait import wait_for
+
+from . import ApplianceException
+from .plugin import AppliancePlugin, AppliancePluginException
+
+
+class ApplianceDBException(AppliancePluginException):
+    pass
+
+
+@attr.s
+class ApplianceDB(AppliancePlugin):
+    _ssh_client = attr.ib()
+
+    @cached_property
+    def client(self):
+        # slightly crappy: anything that changes self.address should also del(self.client)
+        return db.Db(self.address)
+
+    @cached_property
+    def address(self):
+        # pulls the db address from the appliance by default, falling back to the appliance
+        # ip address (and issuing a warning) if that fails. methods that set up the internal
+        # db should set db_address to something else when they do that
+        if self.appliance.db_host:
+            return self.appliace.db_host
+        try:
+            db_addr = self.appliance.wait_for_host_address()
+            if db_addr is None:
+                return self.appliance.address
+            db_addr = db_addr.strip()
+            ip_addr = self.appliance.ssh_client.run_command('ip address show')
+            if db_addr in ip_addr.output or db_addr.startswith('127') or 'localhost' in db_addr:
+                # address is local, use the appliance address
+                return self.appliance.address
+            else:
+                return db_addr
+        except (IOError, KeyError) as exc:
+            self.log.error('Unable to pull database address from appliance')
+            self.log.exception(exc)
+            return self.appliance.address
+
+    @property
+    def is_partition_extended(self):
+        return self.appliance.ssh_client.run_command(
+            "ls /var/www/miq/vmdb/.db_partition_extended") == 0
+
+    @logger_wrap("Extend DB partition")
```

Comment:
I thoguht about it, but a) at the moment they are both using the same one, b) I like the logger wrapper that gives the prefix on all log messages in that unit